File: euler001.py
"""
A solution to Project Euler Problem 1.

https://projecteuler.net/problem=1
"""


def solution(cap: int) -> int:
    """
    Returns the sum of all multiples of 3 or five less than cap

    Examples:
    >>> solution(10) # Problem example
    23
    """
    return sum(i for i in range(cap) if i % 3 == 0 or i % 5 == 0)


if __name__ == '__main__':
    print(solution(1000))

File: go/euler001.go
package main

import "fmt"

const CAP = 1000

func MultiplesSeq(multiples ...int) func() int {
	total := 0
	return func() int {
		for {
			total++
			for _, factor := range multiples {
				if total%factor == 0 {
					return total
				}
			}
		}
	}
}

func main() {
	seq := MultiplesSeq(3, 5)
	sum, i := 0, 0

    for {
        i = seq()
        if i >= CAP {
            break
        }
        sum += i
    }
	fmt.Println(sum)
}
